### ISSUE TITLE
[Feature Request]: Add support for "Preset Positions" feature of "PTZ" cameras

### ISSUE THREAD

**Issue by ekaszubski** [fuatakgun/eufy_security#1119]
**Is your feature request related to a problem? Please describe.**
In addition to pan left/right and tilt up/down, The SoloCam S340 (T8170) and Floodlight Cam E340 (T8425) also support "Preset Positions", which are pre-set camera pointing angles. The app presently allows you to set and command up to five of these positions: a default position, plus four additional positions which are numbered but not (yet) named.

**Describe the solution you'd like**
- The present camera PTZ API has support for pan left/right, tilt up/down, rotate 360, and calibrate: https://github.com/fuatakgun/eufy_security/blob/v8.0.1/custom_components/eufy_security/eufy_security_api/camera.py#L183-L209
- Can we also get support for these "Preset Positions", e.g. Default, Position 1, Position 2, Position 3, Position 4? Perhaps as part of e.g. a new "ptz_position" service?

**Describe alternatives you've considered**
n/a

**Additional context**
The eufy security client project added some CommandType definitions related to this: https://github.com/bropat/eufy-security-client/blob/3.0.0/src/p2p/types.ts#L455-L457 but I don't see any other references to these in their codebase.

I would be happy to assist in reverse-engineering any related parts of the p2p protocol if they are not already known. Any guidance as to where to start (e.g. use wireshark and snoop the packets going to the camera while playing with the app) would be appreciated.

**Comment by awtmarriott** [comment]
Definitely. On a read of the eufy-security-client code these are implemented in the types, interfaces, and implementation of the T8425 floodlight camera already.

**Comment by samso00c** [comment]
That feature would  be great, because I could create an automatism to change camera perspective if motion is detected at another point. Camera needs to turn by 180° from garden to  driveway.

**Comment by Josager** [comment]
It would be fantastic if this feature could be implemented, as it would allow cameras to automatically turn to Location based on time or Motion Sensor Status.

**Comment by ekaszubski** [comment]
@martin-genzel Awesome, thanks! I have a Solocam S340 and 2x Floodlight cam E340 that I could test this on; what is the recommended way to run the test branch? Do I just manually clone it into `config/custom_components/eufy_security`?

Also, should I post any testing-related feedback here or in #1332 ?

**Comment by martin-genzel** [comment]
@ekaszubski I'm not sure what the best practice is here, but manually cloning the files into `config/custom_components/eufy_security` should work (you could also manually copy-paste the modified files if you like). A HA restart is required after that, and please don't forget to create a backup before doing anything 😉

You can test the new services with "Actions" in the [developer tools](https://my.home-assistant.io/redirect/developer_services/). See the PR description for more details.
I'd suggest leaving your feedback in #1332, which makes it easier to incorporate it.

Hope that everything works for you as well 🙂

**Comment by martin-genzel** [comment]
Resolved by #1332.
